R/factorialAnalysis.R

```diff
@@ -183,24 +234,71 @@ factorialAnalysis <- function(jaspResults, dataset, options, ...) {
+    .factorialRegressionANOVAfillTable(factorialRegressionANOVA, options, fit)
+
   jaspResults[["factorialRegressionANOVA"]] <- factorialRegressionANOVA
+
+}
+
+.factorialRegressionANOVAfillTable <- function(factorialRegressionANOVA, options, fit){
+
+  anova <- summary(aov(fit))
+
+  names <- c(names(aov(fit)$coefficients)[-1], "Residuals")
+  anovaFill <- as.data.frame(matrix(0, length(names), 6))
+
+  anovaFill[,1] <- names
+  anovaFill[,2] <- anova[[1]]$Df
+  anovaFill[,3] <- anova[[1]]$`Sum Sq`
+  anovaFill[,4] <- anova[[1]]$`Mean Sq`
+  anovaFill[,5] <- anova[[1]]$`F value`
+  anovaFill[,6] <- anova[[1]]$`Pr(>F)`
+
+  colnames(anovaFill) <- c("terms", "df", "SS", "adjMS", "F", "p")
+
+  factorialRegressionANOVA$setData(anovaFill)
+
+  return()
 }
 
-.factorialRegressionCoefficientsCreateTable <- function(jaspResults, options) {
+.factorialRegressionCoefficientsCreateTable <- function(jaspResults, options, ready, fit){
 
   factorialRegressionCoefficients <- createJaspTable(gettext("Coefficients"))
   factorialRegressionCoefficients$position <- 2
 
-  factorialRegressionCoefficients$dependOn(c("FAassginedFactors",
-                                             "FAresponse"))
+  factorialRegressionCoefficients$dependOn(options = c("FAassginedFactors",
+                                                       "FAresponse",
+                                                       "intOrder"))
 
   factorialRegressionCoefficients$addColumnInfo(name = "terms", title = "", type = "string")
-  factorialRegressionCoefficients$addColumnInfo(name = "effect", title = gettext("Effect"), type = "number")
-  factorialRegressionCoefficients$addColumnInfo(name = "unst", title = gettext("Unstandardized"), type = "number")
+  factorialRegressionCoefficients$addColumnInfo(name = "coef", title = gettext("Coefficient"), type = "number")
   factorialRegressionCoefficients$addColumnInfo(name = "se", title = gettext("Standard Error"), type = "number")
-  factorialRegressionCoefficients$addColumnInfo(name = "st", title = gettext("Standardized"), type = "number")
   factorialRegressionCoefficients$addColumnInfo(name = "t", title = gettext("t"), type = "number")
   factorialRegressionCoefficients$addColumnInfo(name = "p", title = gettext("p"), type = "number")
 
+  if(!is.null(fit))
+    .factorialRegressionCoefficientsFillTable(factorialRegressionCoefficients, options, fit)
+
   jaspResults[["factorialRegressionCoefficients"]] <- factorialRegressionCoefficients
 }
+
+.factorialRegressionCoefficientsFillTable <- function(factorialRegressionCoefficients, options, fit){
+
+  coefs <- as.data.frame(summary(fit)$coefficients)
+
+  names <- names(coef(fit))
+  coefsFill <-  as.data.frame(matrix(0, length(names), 5))
+
+  coefsFill[,1] <- gettext(names)
+  coefsFill[,2] <- coefs$Estimate
+  coefsFill[,3] <- coefs$`Std. Error`
+  coefsFill[,4] <- coefs$`t value`
+  coefsFill[,5] <- coefs$`Pr(>|t|)`
+
+  colnames(coefsFill) <- c("terms", "coef", "se", "t", "p")
```

Comment:
here you can just use `data.frame` directly again. Also `gettext(names)` doesn't do anything, you can delete the `gettext`.

Reply:
✅